src/SyGuS.ml

```diff
@@ -131,8 +124,19 @@ let parse_sexps (sexps : Sexp.t list) : t =
               | List [ (Atom "inv-constraint") ; (Atom _invf_name) ; (Atom _pref_name)
                                                ; (Atom _transf_name) ; (Atom _postf_name) ]
                 -> pref_name := _pref_name ; transf_name := _transf_name ; postf_name := _postf_name
-                 ; if not (String.equal !invf_name _invf_name)
-                   then raise (Parse_Exn ("Invariant function [" ^ _invf_name ^ "] not declared"))
+                 ; (if not (String.equal !invf_name _invf_name)
+                    then raise (Parse_Exn ("Invariant function [" ^ _invf_name ^ "] not declared")))
+                 ; (if not (String.equal !pref_name _pref_name)
+                    then raise (Parse_Exn ("Precondition function [" ^ _pref_name ^ "] not declared")))
+                 ; (if not (String.equal !transf_name _transf_name)
+                    then raise (Parse_Exn ("Transition function [" ^ _transf_name ^ "] not declared")))
+                 ; (if not (String.equal !postf_name _postf_name)
+                    then raise (Parse_Exn ("Postcondition function [" ^ _postf_name ^ "] not declared")))
```

Comment:
These 4 checks seem incorrect. Right before the checks (at line 126) we set `pref_name := _pref_name` and so on, so it would never be the case that `not (String.equal !pref_name _pref_name)`.

It only makes sense to check the `invf_name` since that function name has to match at `inv-constraint` (where `_invf_name` is set`) and `synth-inv` (where `invf_name` is set).

Reply:
What I meant was that you should check if `_pref_name`, `_transf_name` and `_postf_name` exist in the set of functions (`funcs`).

Reply:
You should remove these 3 checks you added.

They can be merged with the `check_arg_order` checks. See my comment below.